### PR TITLE
Enhance the StyleWrapper classNames generator with generic pluggable extenders

### PR DESCRIPTION
**Pull request by sneridagh**
We've been using this to fine adjust the vertical spacing in our last projects. I had doubts about to push this for core, but I decided that could be beneficial for Volto. The drawback is that they could add quite a lot of classnames to the block, but still, my experience and feedback is quite good. Any help that could come from the code in order to style the blocks is welcomed.

### CODE REVIEW

**Review by tiberiuichim — COMMENTED** [review]
I think we should run the extenders as "pipe filters" rather then combining their output at the end. 

Simplified, the signature of an extender should be `({classnames, data, ...}) => classnames`. It should get a list of classnames and return a list of classnames. This allows the extender to make decisions such as, for example, not adding its classname because something else was added before, or avoiding duplication.

**Comment by sneridagh** [comment]
@tiberiuichim I think it's implemented as you commented, soryy I didn't updated the docs, I will do. Take a look at the tests.

**Review by tiberiuichim — APPROVED** [review]

**Comment by sneridagh** [comment]
@ksuess @tiberiuichim please review again!

**Review by ksuess — APPROVED** [review]
Perfect! Great extension.

File: docs/source/configuration/settings-reference.md [hunk 1]
```diff
@@ -232,6 +232,71 @@ styleClassNameConverters
     data to actual class names. You can customize the generated classname by
     registering fieldnames with names such as `<fieldname>:<converterName>`,
     where the converter is registered here.
+
+styleClassNameExtenders
+    An array containing functions that extends how the StyleWrapper builds a list of styles. These functions have the signature `({ block, content, data, classNames }) => classNames`. Here some examples of useful ones, for simpliciy, they are compacted in one extender:
```
Comment (davisagli) [inline docs/source/configuration/settings-reference.md:237]:
```suggestion
    An array containing functions that extends how the StyleWrapper builds a list of styles. These functions have the signature `({ block, content, data, classNames }) => classNames`. Here some examples of useful ones, for simplicity, they are compacted in one extender:
```

File: src/components/manage/Blocks/Block/StyleWrapper.jsx [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 const StyleWrapper = (props) => {
-  const { children, data = {} } = props;
+  const { children, content, data = {}, block } = props;
   const styles = buildStyleClassNamesFromData(data.styles);
+
+  const styleExtenders = buildStyleClassNamesExtenders({
+    block,
+    content,
+    data,
+  });
```
[the remark this answers is not in the document]
Comment (tiberiuichim) [reply src/components/manage/Blocks/Block/StyleWrapper.jsx]: I think we should keep data as passed prop.

**Comment by sneridagh** [comment]
@davisagli crap after merge GH updated the page with your comment! I will manually fix it!

**Comment by sneridagh** [comment]
typo: https://github.com/plone/volto/pull/4268

**Pull request merged**